crates/ra_hir_expand/src/lib.rs

```diff
@@ -129,31 +131,59 @@ impl salsa::InternKey for MacroCallId {
 
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
 pub struct MacroDefId {
-    pub krate: CrateId,
-    pub ast_id: AstId<ast::MacroCall>,
+    // FIXME: krate and ast_id are currently optional because we don't have a
+    // definition location for built-in derives. There is one, though: the
+    // standard library defines them. The problem is that it uses the new
+    // `macro` syntax for this, which we don't support yet. As soon as we do
+    // (which will probably require touching this code), we can instead use
+    // that (and also remove the hacks for resolving built-in derives).
+    pub krate: Option<CrateId>,
+    pub ast_id: Option<AstId<ast::MacroCall>>,
```

Comment:
Note that `proc-macros` don't have location though. In general, one day we'll move to a model where all sources are optional (think using `rlib`s to build HIR).

Reply:
Yeah, proc macros will have some other kind of data for identification, I guess... That's basically the point, we'll have to change this further anyway and I just wanted to do the minimal change for now ;)